apps/service_gather/lib/gather/extraction/source_stream.ex

```diff
@@ -1,58 +1,98 @@
-defmodule Gather.Extraction.SourceStream do
-  defmodule SourceHandler do
-    @moduledoc """
-    Callbacks for handling data messages.
-
-    See [Source.Handler](../../../../protocol_source/lib/source/handler.ex)
-    for more.
-    """
-    use Source.Handler
-    use Properties, otp_app: :service_gather
-
-    getter(:dlq, default: Dlq)
-
-    def handle_batch(batch, context) do
-      send(context.assigns.pid, {:source_batch, batch})
-      :ok
-    end
+defmodule Gather.Extraction.SourceHandler do
+  @moduledoc """
+  Callbacks for handling data messages.
 
-    def send_to_dlq(dead_letters, _context) do
-      dlq().write(dead_letters)
-    end
+  See [Source.Handler](../../../../protocol_source/lib/source/handler.ex)
+  for more.
+  """
+  use Source.Handler
+  use Properties, otp_app: :service_gather
+  use Annotated.Retry
+  require Logger
+
+  @max_tries get_config_value(:max_tries, default: 10)
+  @initial_delay get_config_value(:initial_delay, default: 500)
+
+  getter(:dlq, default: Dlq)
+  getter(:app_name, required: true)
+
+  @impl Source.Handler
+  def handle_batch(batch, %{assigns: %{extract: extract}} = context) do
+    Decoder.decode(extract.decoder, [batch])
```

Comment:
Why wrap the batch in another list? Shouldn't a batch be a list? Should this be `List.wrap` instead?

Reply:
Good callout; in practice these batches were single messages so we were getting "protocol not implemented" warnings for the `Enumerable` but `List.wrap` is a better approach.

Reply:
Won't this go away once we refactor the Decoder protocol

Reply:
I don't want this to wait on the decoder so i like the wrap approach and removing it if necessary can be part of the final implementation of the decoder refactor.